### PR TITLE
Updates legacy training to save in global checkpoints directory

### PR DESCRIPTION
**Pull request by RobotSail**
Signed-off-by: Oleg S <97077423+RobotSail@users.noreply.github.com>

Currently, when we train models we always save in a directory relative to the caller called './models'.
This PR updates the legacy training code to save checkpoints into the global checkpoints directory.

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [x] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Review by russellb — COMMENTED** [review]
would be great to see `e2e-nvidia-a10g-x1.yml` pass on this, since that's the one that exercises this code path.

The e2e that runs automatically exits before doing the model conversion and output into this final directory, because that doesn't work with `--4-bit-quant` on

**Comment by RobotSail** [comment]
@russellb I just kicked it off, let's see if it works. My prediction is it will fail because I believe there are a few other issues in the current e2e.

File: CHANGELOG.md [hunk 1]
```diff
@@ -53,6 +53,11 @@
 * `ilab model chat`: Chatlogs are now stored under the `instructlab/checkpoints` directory in the
    platform's dedicated data directory under the `instructlab` package.
 
+### Fixes
```
[the remark this answers is not in the document]
Comment (RobotSail) [reply CHANGELOG.md:56]: @russellb Sure, I'll move it there.

**Review by leseb — COMMENTED** [review]
I'm not entirely sure I grasp all the implications of this change, so I'll leave the approval to someone with more expertise. However, it looks good to me (LGTM).

**Comment by nathan-weinberg** [comment]
> I'm not entirely sure I grasp all the implications of this change, so I'll leave the approval to someone with more expertise. However, it looks good to me (LGTM).

From what I understand (@RobotSail please correct me if I am wrong :smile:) the implications are this:
- Prev, we only had "legacy" training which would save newly-trained models to the `models/` directory
- "New" training saves newly-trained model to the new storage directory Oleg implemented in #1471 
- This change is making it so legacy training saves new models to the same directory as new training

**Comment by RobotSail** [comment]
That's correct @nathan-weinberg.

**Review by booxter — APPROVED** [review]
LGTM; Russell suggestion for the changelog should probably be followed.

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (RobotSail) [reply CHANGELOG.md]: Done

**Comment by RobotSail** [comment]
@booxter Done, thank you for the reminder.

**Review by booxter — APPROVED** [review]
I am approving but if you need to respin this, please re-wrap the commit message so that it fits in 72 characters (as required for smooth operation of tools like `git log`.) Thanks.

**Review by n1hility — APPROVED** [review]

**Comment by booxter** [comment]
The patch is fine, but

> if you need to respin this, please re-wrap the commit message so that it fits in 72 characters (as required for smooth operation of tools like git log.) Thanks.

Would you mind since you just respinned?

**Review by cdoern — APPROVED** [review]

**Pull request merged**